modules/UI/videolayout/SmallVideo.js

```diff
@@ -520,49 +520,40 @@ SmallVideo.prototype.isVideoPlayable = function() {
  * @return {number} one of <tt>DISPLAY_VIDEO</tt>,<tt>DISPLAY_AVATAR</tt>
  * or <tt>DISPLAY_BLACKNESS_WITH_NAME</tt>.
  */
-SmallVideo.prototype.selectDisplayMode = function() {
-    const isAudioOnly = APP.conference.isAudioOnly();
-    const tileViewEnabled = shouldDisplayTileView(APP.store.getState());
-    const isVideoPlayable = this.isVideoPlayable();
-    const hasVideo = Boolean(this.selectVideoElement().length);
+SmallVideo.prototype.selectDisplayMode = function(input) {
 
     // Display name is always and only displayed when user is on the stage
-    if (this.isCurrentlyOnLargeVideo() && !tileViewEnabled) {
-        return isVideoPlayable && !isAudioOnly ? DISPLAY_BLACKNESS_WITH_NAME : DISPLAY_AVATAR_WITH_NAME;
-    } else if (isVideoPlayable && hasVideo && !isAudioOnly) {
+    if (input.isCurrentlyOnLargeVideo && !input.tileViewEnabled) {
+        return input.isVideoPlayable && !input.isAudioOnly ? DISPLAY_BLACKNESS_WITH_NAME : DISPLAY_AVATAR_WITH_NAME;
+    } else if (input.isVideoPlayable && input.hasVideo && !input.isAudioOnly) {
         // check hovering and change state to video with name
-        return this._isHovered() ? DISPLAY_VIDEO_WITH_NAME : DISPLAY_VIDEO;
+        return input.isHovered ? DISPLAY_VIDEO_WITH_NAME : DISPLAY_VIDEO;
     }
 
     // check hovering and change state to avatar with name
-    return this._isHovered() ? DISPLAY_AVATAR_WITH_NAME : DISPLAY_AVATAR;
+    return input.isHovered ? DISPLAY_AVATAR_WITH_NAME : DISPLAY_AVATAR;
 };
 
 /**
- * Prints information about the current display mode.
+ * Computes information that determine the display mode.
  *
- * @param {string} mode - The current mode.
- * @returns {void}
- */
-SmallVideo.prototype._printDisplayModeInfo = function(mode) {
-    const isAudioOnly = APP.conference.isAudioOnly();
-    const tileViewEnabled = shouldDisplayTileView(APP.store.getState());
-    const isVideoPlayable = this.isVideoPlayable();
-    const hasVideo = Boolean(this.selectVideoElement().length);
-    const displayModeInfo = {
-        isAudioOnly,
-        tileViewEnabled,
-        isVideoPlayable,
-        hasVideo,
+ * @returns {Object}
+ */
+SmallVideo.prototype.computeDisplayModeInput = function() {
+    return {
+        isCurrentlyOnLargeVideo: this.isCurrentlyOnLargeVideo(),
+        isHovered: this._isHovered(),
+        isAudioOnly: APP.conference.isAudioOnly(),
```

Comment:
Can you please fold this into the isVIdeoPlayable test, while we're here?

Reply:
Will do! And in that case, shall I also remove the `this.videoStream.isMuted()` check as per our discussion earlier today?

Reply:
(I'll test these before I push a new commit)

Reply:
Yeah, let's do that!

Reply:
Just pushed a new commit. I tested it in a bunch of different scenarios and it seems to work fine.